src/main/java/org/aksw/simba/lemming/IColouredGraph.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * 
+ */
+package org.aksw.simba.lemming;
+
+import com.carrotsearch.hppc.BitSet;
+
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntSet;
+
+/**
+ * This interface is used for the decorator pattern to simplify multithreading
+ * in the graph amendment phase
+ * 
+ * @author Pranav
```

Comment:
This is not really a good description of the interface and its usage :wink:

Reply:
Improved the javadoc